app/controllers/base_metric_configurations_controller.rb

```diff
@@ -1,42 +1,147 @@
   before_action :metric_configuration_owner?, only: [:edit, :update, :destroy]
   before_action :kalibro_configuration_owner?, only: [:new, :create, :choose_metric]
-  before_action :set_metric_configuration, only: [:show, :edit, :update, :destroy]
+  before_action :set_kalibro_configuration!
+  before_action :find_metric_configuration!, only: [:show, :edit, :update, :destroy]
+  before_action :new_metric_configuration!, only: [:create]
+  before_action :set_metric!, only: [:create, :update]
+  before_action :set_reading_group!, only: [:show, :edit, :create, :update]
+
+  def show
+    @kalibro_ranges = @metric_configuration.kalibro_ranges
+  end
 
   def new
-    update_metric_configuration(MetricConfiguration.new)
+    @metric_configuration = MetricConfiguration.new
   end
 
-  def show
-    if metric_configuration
-      @reading_group = ReadingGroup.find(metric_configuration.reading_group_id)
-      @kalibro_ranges = metric_configuration.kalibro_ranges
-    else
-      raise NotImplementedError
-    end
+  def edit
   end
 
   def create
-    update_metric_configuration(MetricConfiguration.new(metric_configuration_params))
+    respond_to { |format| save_and_redir(format) }
+  end
+
+  def update
+    respond_to do |format|
+      save_and_redir(format) do |metric_configuration|
+        metric_configuration.update(metric_configuration_params)
+      end
+    end
+  end
+
+  def destroy
+    @metric_configuration.destroy
+    clear_caches
+
+    respond_to do |format|
+      format.html { redirect_to kalibro_configuration_path(@kalibro_configuration.id) }
+      format.json { head :no_content }
+    end
   end
 
   protected
 
-  def metric_configuration
-    raise NotImplementedError
+  def save_and_redir(format)
+    new_record = @metric_configuration.id.nil?
+    result = block_given? ? (yield @metric_configuration) : @metric_configuration.save
+
+    if result
+      clear_caches
+
+      format.html do
+        redirect_to kalibro_configuration_path(@kalibro_configuration.id),
+          notice: t(new_record ? 'successfully_created' : 'successfully_updated',
+                    record: t(@metric_configuration.class))
+      end
+      format.json { render json: @metric_configuration, status: new_record ? :created : :ok }
+    else
+      failed_action(format)
+    end
   end
 
-  def update_metric_configuration (new_metric_configuration)
-    raise NotImplementedError
+  def failed_action(format, error = nil)
+    errors = @metric_configuration.kalibro_errors
+    errors << error unless error.nil?
+
+    flash[:notice] = errors.join(', ')
+
+    format.json { render json: { errors: errors }, status: :unprocessable_entity }
+    render_failure_html(format)
   end
 
-  # Never trust parameters from the scary internet, only allow the white list through.
-  # TODO: this should be refactored to the concern metric configuration
+  def render_failure_html(format)
+    if action_name == 'create'
```

Comment:
I'm the one that wrote this, but now thinking about it it does seem bad to check the action name here. Unfortunately I can't think of a great solution: my other choice would be to do a `redirect_to :back` and try to handle an exception for when the request has no referrer, but I believe it doesn't preserve the post parameters, which would empty forms of the `new` page in case of errors, which is terrible UX.